test/abe/dippe.c

```diff
@@ -0,0 +1,214 @@
+
+#include <amcl/pair_BN254.h>
+
+#include "cifer/test.h"
+#include "cifer/data/vec.h"
+#include "cifer/abe/dippe.h"
+
+MunitResult test_dippe_end_to_end_conjunction(const MunitParameter *params, void *data) {
+    cfe_error err;
+
+    // Init DIPPE scheme
+    cfe_dippe dippe;
+    cfe_dippe_setup(&dippe, 2);
+
+    // Setup two Authorities
+    cfe_dippe_pub_key pk[2];
+    cfe_dippe_sec_key sk[2];
+    cfe_dippe_authsetup(&dippe, &(pk[0]), &(sk[0]));
+    cfe_dippe_authsetup(&dippe, &(pk[1]), &(sk[1]));
+
+    // Sample message
+    ECP_BN254 g1;
+    ECP_BN254_generator(&g1);
+    ECP2_BN254 g2;
+    ECP2_BN254_generator(&g2);
+    FP12_BN254 msg;
+    PAIR_BN254_ate(&msg, &g2, &g1);
+    PAIR_BN254_fexp(&msg);
+
+    // Collection of 6 public keys
+    // Each of them correspondents to an index of the Attribute-/ Policy-vector
+    cfe_dippe_pub_key *pks[] = {&(pk[0]), &(pk[1]), &(pk[0]), &(pk[1]), &(pk[0]), &(pk[1])};
+
+    // unique GID
+    char gid[] = "TESTGID";
```

Comment:
Shouldn't each user get its own unique GID?

Reply:
That's right. But the test case here only simulates a single user. For each of the attribute vector components, the user has to request a decryption key from the respective authority. The GID for all requests has to be the same, so that the resulting keys fit together.

Reply:
Aren't you simulating multiple users, each with a different set of attributes? Because in your interpretation the one user chooses different vectors as if he can choose which attributes he has? Maybe I am wrong...

Reply:
Ah yea, now I get what you mean. I thought of it as independent test runs for a single user but your interpretation makes more sense. I changed it, so that the GID is modified a bit for each run.